src/base/perf_monitor_trackers.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * perf_monitor_trackers.hpp  - helper c++ classes over c code that ease tracking of resources
+ */
+
+#ifndef PERF_MONITOR_HPP
+#define PERF_MONITOR_HPP
+
+#include "perf_monitor.h"
+
+/* manual perf stat monitoring utility for counter timer values
+ *
+ * NOTE: the global 'perfmon_is_perf_tracking' is only checked in the ctor
+ * and used for the rest of the lifetime as is
+ */
+class perfmon_manual_tracker_counter_timer
```

Comment:
Can it be named without "manual"? Also, "perfmon counter timer tracker" sounds better (more natural in English)
```suggestion
class perfmon_counter_timer_tracker (PERF_STAT_ID a_stat_id);
```

Shouldn't default constructor be deleted?

Reply:
Renamed. I think the default ctor is automatically dtor once there is a user defined one. I deleted the other ctors and ops.